portal-api/src/v1/users/passport.js

```diff
@@ -39,28 +49,45 @@ const baseAuthenticationConfiguration = ({ name, passport, additionalValidation,
           return done(error, false);
         }
 
-        const validation = additionalValidation
-          ? user && user.sessionIdentifier === jwtPayload.sessionIdentifier && additionalValidation(jwtPayload)
-          : user && user.sessionIdentifier === jwtPayload.sessionIdentifier;
+        if (sessionIdentifierValidation(user, jwtPayload) === PASSPORT_VALIDATION_RESULTS.FAILED) {
+          console.error("Failed JWT validation for '%s' strategy", name);
+          return done(null, false);
+        }
 
-        if (validation) {
-          const additionalFields = getAdditionalReturnedFields(user);
-          return done(null, { ...sanitize(user), ...additionalFields });
+        if (additionalUserValidation && additionalUserValidation(user, jwtPayload) === PASSPORT_VALIDATION_RESULTS.FAILED) {
```

Comment:
updated validation in passport so that if we provide extra user validation we provide a clear console log and don't authenticate if they are blocked or disabled once they are on the login-complete passport flow